src/utils/getFiltered.ts

```diff
@@ -0,0 +1,64 @@
+import { Person } from '../types';
+import { Sort } from '../types/Sort';
+
+export function getFilteredPeople(
+  people: Person[],
+  {
+    sort,
+    order,
+    query,
+    sex,
+    centuries,
+  }: {
+    sort: string | null;
+    order: string | null;
+    query: string;
+    sex: string;
+    centuries: string[];
+  },
+) {
+  let filteredPeople = [...people];
+  const normilizeQuery = query.toLowerCase();
+
+  if (sort) {
+    filteredPeople.sort((p1, p2) => {
+      switch (sort) {
+        case Sort.Name:
+        case Sort.Sex:
+          return p1[sort].localeCompare(p2[sort]);
+        case Sort.Born:
+        case Sort.Died:
+          return p1[sort] - p2[sort];
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (order) {
+    filteredPeople.reverse();
+  }
+
+  if (query) {
+    filteredPeople = filteredPeople.filter(
+      p =>
+        p.name.toLowerCase().includes(normilizeQuery) ||
+        (p.motherName && p.motherName.toLowerCase().includes(normilizeQuery)) ||
+        (p.fatherName && p.fatherName.toLowerCase().includes(normilizeQuery)),
+    );
+  }
+
+  if (sex) {
+    filteredPeople = filteredPeople.filter(p => p.sex === sex);
+  }
```

Comment:
The 'sex' filter checks for exact matches. This could lead to incorrect results if the 'sex' parameter is not exactly the same as the 'sex' property of the person objects. It would be better to use a case-insensitive comparison.